lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -51,10 +23,76 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def serializable_hash_for_collection(options = {})
+          serializer.map { |s| Attributes.new(s, instance_options).serializable_hash(options) }
+        end
+
+        def serializable_hash_for_single_resource(options = {})
+          resource = resource_object_for(serializer, options)
+          resource.merge!(resource_relationships)
+        end
+
+        # iterate through the associations on the serializer,
+        # adding them to the parent as needed (as singular or plural)
+        #
+        # nested_associations is a list of symbols that governs what
+        # associations on the passed in seralizer to include
+        def resource_relationships
+          relationships = {}
+
+          serializer.associations(@include_tree).each do |association|
+            association_serializer = association.serializer
+            opts = association.options
+            key = association.key
+            opts[:include] = @include_tree[key].to_hash
```

Comment:
Also, the association's `options` is probably not the right place to pass the include_tree. It should be in the adapter `options`.